src/main/java/games/strategy/triplea/delegate/DependentBattle.java

```diff
@@ -0,0 +1,44 @@
+package games.strategy.triplea.delegate;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+
+/**
+ * Battle with possible dependencies
+ * Includes MustFightBattle and NonFightingBattle
+ */
+abstract public class DependentBattle extends AbstractBattle {
```

Comment:
Please change this to `public abstract class` in order to be consequent in terms of naming...